.gitignore

```diff
@@ -34,6 +34,9 @@ share/python-wheels/
 *.egg
 MANIFEST
 
+# Version created and populated by setuptools_scm
+_version.py
```

Comment:
I'd prefer a `.gitignore` file in the folder where this is needed. The current entry as added right now excludes any file with this name. So we'd need to use a full path to be precise in what we ignore if it stays in this file.

Although, it probably should go to the end of the file and not be in the middle since it seems to be partially generated from https://gitignore.io, and putting things in the middle of the template makes it harder to update it or track the boilerplate separately from the projec-specific entries.

Still, I'd prefer a gitignore in a more accurate location.

Reply:
It was added here, directly after build excludes because it is related to the build.

It is not uncommon for it to be in the .gitignore:

https://github.com/ansible/pylibssh/blob/devel/.gitignore#L398

https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/master/.gitignore#L132

https://github.com/ansible-community/ansible-lint/blob/main/.gitignore#L50

Nor it is always at the bottom.

Reply:
> It was added here, directly after build excludes because it is related to the build.

I wish it was structured better. Ideally, the boilerplate should go first and have start/end markers.

> It is not uncommon for it to be in the .gitignore:

You're right. Yet this doesn't mean that it's not handy to have a dedicated file without a boilerplate: we have one right in this project, in the changelog fragments directory, for example.

Still, if you want to have it here, at least be precise about which one to exclude:
```suggestion
/src/ansible_navigator/_version.py
```

> [ansible/pylibssh@`devel`/.gitignore#L398](https://github.com/ansible/pylibssh/blob/devel/.gitignore?rgh-link-date=2022-02-21T18%3A40%3A42Z#L398)
> 
> [sphinx-contrib/sphinxcontrib-towncrier@`master`/.gitignore#L132](https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/master/.gitignore?rgh-link-date=2022-02-21T18%3A40%3A42Z#L132)
> 
> [ansible-community/ansible-lint@`main`/.gitignore#L50](https://github.com/ansible-community/ansible-lint/blob/main/.gitignore?rgh-link-date=2022-02-21T18%3A40%3A42Z#L50)
> 
> Nor it is always at the bottom.

It is, in all of the examples you've linked above. The `/.test-results/` entry is also project-specific which is why it shares the "bottom". In some places, there's a marker that shows where the boilerplate ends, like `# End of https://www.gitignore.io/api/git,linux,pydev,python,windows,pycharm+all,jupyternotebook,vim,webstorm,emacs,dotenv` but others don't have it for some reason.

Reply:
I am going to move it, brb

Reply:
ok, I prefer that, indicates it is there for a good reason and not a cut and paste from ?

Reply:
Maybe we should generate some boilerplate from https://www.toptal.com/developers/gitignore at some point and replace bits of the old one (as a separate effort) — it emits those markers automatically.